components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -945,9 +945,20 @@ class BraveRewardsBrowserTest
   }
 
   void RefreshPublisherListUsingRewardsPopup() const {
-    rewards_service_browsertest_utils::WaitForElementThenClick(
-        OpenRewardsPopup(),
-        "[data-test-id='unverified-check-button']");
+    // Refresh publisher list; assume that absence of refresh button
```

Comment:
how come publisher is already verified if we didn't click refresh yet?

Reply:
From my understanding of the code the publisher list also updates on a timer, so sometimes the manual click isn't needed.

Reply:
interesting so you thing that timer is causing problems

Reply:
Yeah, the timer is kicking off the update and updating the server before there's any chance to click the button (at least on my local Windows machine). Once the rewards popup opens, there's no longer a "Refresh Status" button to click on because it's already updated to a verified publisher.

Reply:
this happened because we don't send any end when doing pagination, so pagination will continue for 100 pages and because of that we have this race condition